### PR TITLE
feat: middleware.ts created for protect the routes

### PR DESCRIPTION
**Pull request by black-arm**

### CODE REVIEW

**Comment by Balastrong** [comment]
Thank you for the PR!

When I navigate to a protected page I now get this error, we should probably make sure it redirects to the homepage. Any idea?

image

**Comment by black-arm** [comment]
the redirect work if we deleted secret variable from [...nextauth].ts and in .env.local  we added the NEXTAUTH_SECRET

Screenshot 2023-08-25 alle 09 58 04
Screenshot 2023-08-25 alle 10 01 40
I check 

how do we want to proceed? Do we update the .env and documentation or do we find to way to work with secret variable

**Comment by Balastrong** [comment]
Finding a way that does not require an extra variable would be great :)

**Comment by Balastrong** [comment]
Still thinking about it... we might entirely replace `SECRET` with `NEXTAUTH_SECRET` and leave a value there already in `.env.example` to keep onboarding easy

**Comment by Balastrong** [comment]
Exactly, and you can also give a default value to `NEXTAUTH_SECRET` in `.env.example` so that it just works :)

File: .env.example [hunk 1]
```diff
@@ -6,4 +6,4 @@ DEV_GITHUB_TOKEN=
 #GITHUB_SECRET=
 
 # Next-Auth secret (not required for local development)
```
Comment (Balastrong) [inline .env.example:8]: The description should be updated too as now it's actually required to make it work

File: .env.example [hunk 2]
```diff
@@ -6,4 +6,4 @@ DEV_GITHUB_TOKEN=
...
-#SECRET=
\ No newline at end of file
+#NEXTAUTH_SECRET=oYoOxbxGQmKOqbJmxf5h1RScYrC8DZ2BgL2OxT5w/C8=
```
Comment (Balastrong) [inline .env.example:9]: Since it is now mandatory, I'd remove the comment `#` so it can already be used

**Review by Balastrong — CHANGES_REQUESTED** [review]
Thank you!

I added a couple comments to make sure the onboarding remains correct and seamless for new contributors

**Review by Balastrong — APPROVED** [review]
LGTM! Thank you! 🚀

**Pull request merged**